forEach/challenges-01.test.js

```diff
@@ -0,0 +1,280 @@
+'use strict';
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 1
+//
+// Write a function named pushIntoSecond which takes in an array and uses a for loop
+// to push all of the elements from the initial array into a second array. 
+// 
+// Return the second array.
+// ------------------------------------------------------------------------------------------------
+
+const pushIntoSecond = (arr) => {
+  const second = [];
+  for (let i = 0; i < arr.length; i++) {
+    second.push(arr[i]);
+  }
+  return second;
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 2
+//
+// Write a function named addBonusPoints that takes in an array of raw test scores
+// and adds five bonus points to each score.
+//
+// Use the for...of syntax.
+// 
+// Return an array of scores that have had the bonus points added.
+// ------------------------------------------------------------------------------------------------
+
+const addBonusPoints = (arr) => {
+  let arrFinalScores = [];
+  for (let value of arr) {
+    arrFinalScores.push(value + 5);
+  }
+  return arrFinalScores;
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 3
+//
+// Write a function named addCurve that takes in an array of raw test scores and increases each score by 5%.
+// 
+// Use the for...in syntax.
+//
+// Return an array of curved scores;
+// ------------------------------------------------------------------------------------------------
+
+const addCurve = (arr) => {
+  let arrFinalScores = [];
+  for (let index in arr) {
+    arrFinalScores.push(arr[index] * 1.05);
+  }
+  return arrFinalScores;
+}
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 4
+//
+// Write a function named greeting that takes in a string and returns the string in all uppercase letters. 
+// 
+// Then, write a function named speaker that takes in a string and a callback function. 
+// The speaker function should return the string in all uppercase letters only by invoking the callback.
+// ------------------------------------------------------------------------------------------------
+
+const greeting = (word) => {
+  return word.toUpperCase();
+};
+
+const speaker = (message, callback) => {
+  return callback(greeting(message));
+};
+

```

Comment:
so the greeting() is actually what will be passed in as the callback parameter so you shouldn't specify it in your code.
```
const speaker = (message, callback) => {
  // Solution code here...
  return callback(message);
};
```